lib/Dialect/TTIR/Transforms/Passes.cpp

```diff
@@ -614,14 +614,25 @@ createToLayoutOp(PatternRewriter &rewriter, Location loc, Value input,
   auto desiredLayout =
       rewriter.getAttr<LayoutAttr>(ty, desiredMemorySpace, currLayout.getGrid(),
                                    desiredElementType, desiredMemLayout);
+
+  ::mlir::TypedValue<::mlir::RankedTensorType> existingOpType = nullptr;
+  tensor::EmptyOp existingEmpty = input.getDefiningOp<tensor::EmptyOp>();
+  ttir::ConstantOp existingConstant = input.getDefiningOp<ttir::ConstantOp>();
+  if (existingEmpty) {
+    existingOpType = existingEmpty.getResult();
+  } else if (existingConstant) {
+    existingOpType = existingConstant.getResult();
+  }
+
+  if (existingOpType) {
+    existingOpType.setType(mlir::RankedTensorType::get(
+        ty.getShape(), ty.getElementType(), desiredLayout));
+    return existingOpType;
```

Comment:
I'm not 100% sure it's safe to modify the op in place using PatternRewriter, I think you need to wrap this in:
```c++
rewriter.modifyOpInPlace(...
```

Or it might be clearer to write:
```c++
return rewriter.replaceOpWithNewOp(
  exising_empty,
  ty.getShape(),
  ty.getElementType(),
  desiredLayout).getResult();
```

Reply:
Oh right, I forgot the wrapper.

I opted for changing the output type instead of replacing ops, because constructor for each op is different, for example constructor for `tensor.empty` op accepts layout encoding as parameter, while constructor for `ttir.constant` op doesn't so we have to construct type with that encoding and pass it in. Currently we only support those two tensor producer ops, but maybe we will have more in the future, so I thought to make that code more uniform, and since they all return `RankedTensorType` I opted for just getting the type and changing it. We also do that at the end of `matchAndRewrite` function inside `TTIRLayoutDPSOperandsRewriter`:
```
        rewriter.modifyOpInPlace(op, [&]() {
          modified = true;
          op->setOperand(operand.getOperandNumber(), *desiredLayout);
          if (isResult) {
            // If this is the output operand, update the result type
            op->getResult(0).setType(desiredLayout->getType());
          }
        });
```

If you still think it would be cleaner to replace with new ops lmk I will do it, I don't have a strong preference here.

Reply:
I slightly prefer replace, I find it to be more readable, but I will not hold up the review, it's a minor nit.

Reply:
`modifyOpInPlace` takes op as argument so there goes my uniformity goal :) I will change to replace with new ops.